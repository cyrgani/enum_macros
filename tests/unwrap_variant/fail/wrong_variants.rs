use enum_macros::UnwrapVariant;

#[derive(UnwrapVariant)]
#[allow(dead_code)]
enum Test {
    #[unwrap(ref)]
    A { v: u8 },
    #[unwrap(mut)]
    B,
    #[unwrap(ref, mut)]
    C(u8, u8),
    #[unwrap(ref)]
    D { },
}

fn main() {}