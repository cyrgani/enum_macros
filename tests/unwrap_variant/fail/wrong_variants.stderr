error: UnwrapVariant can only be derived for single unnamed variants
 --> tests/unwrap_variant/fail/wrong_variants.rs:6:5
  |
6 | /     #[unwrap(ref)]
7 | |     A { v: u8 },
  | |_______________^

error: UnwrapVariant can only be derived for single unnamed variants
 --> tests/unwrap_variant/fail/wrong_variants.rs:8:5
  |
8 | /     #[unwrap(mut)]
9 | |     B,
  | |_____^

error: UnwrapVariant can only be derived for single unnamed variants
  --> tests/unwrap_variant/fail/wrong_variants.rs:10:5
   |
10 | /     #[unwrap(ref, mut)]
11 | |     C(u8, u8),
   | |_____________^

error: UnwrapVariant can only be derived for single unnamed variants
  --> tests/unwrap_variant/fail/wrong_variants.rs:12:5
   |
12 | /     #[unwrap(ref)]
13 | |     D { },
   | |_________^
